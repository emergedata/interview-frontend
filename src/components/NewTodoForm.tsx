import { Button, Paper, Stack, TextField, Typography } from "@mui/material";
import { addTodo } from "../utils/data";
import { useState } from "react";

export const NewTodoForm = () => {
  const [addTodoName, setAddTodoName] = useState("");
  const [addTodoDesc, setAddTodoDesc] = useState("");

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    addTodo({ name: addTodoName, details: addTodoDesc });
    setAddTodoName(""); // added to clear form
    setAddTodoDesc(""); // added to clear form
  };

  return (
    <Paper elevation={2}>
      <form onSubmit={handleSubmit}>
        <Stack
          sx={{
            width: 200,
            margin: 2,
            gap: 1,
          }}
        >
          <Typography fontSize={20}>Add a todo</Typography>
          <TextField
            id="name"
            label="Name"
            value={addTodoName}
            onChange={(e) => setAddTodoName(e.target.value)}
          />
          <TextField
            id="desc"
            label="Description"
            value={addTodoDesc}
            onChange={(e) => setAddTodoDesc(e.target.value)}
          />
          <Button type="submit">Add</Button>
        </Stack>
      </form>
    </Paper>
  );
};
