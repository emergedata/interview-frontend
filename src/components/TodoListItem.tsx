import { ListItem, ListItemText } from "@mui/material";
import { Todo } from "../models/todo";

export const TodoListItem = ({ todo }: { todo: Todo }) => (
  <ListItem>
    <ListItemText primary={todo.name} secondary={todo.details} />
  </ListItem>
);
