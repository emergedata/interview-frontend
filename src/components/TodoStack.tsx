import { Stack } from "@mui/material";
import { Todo } from "../models/todo";
import { TodoListItem } from "./TodoListItem";


export const TodoStack = ({ todos }: { todos: Todo[]; }) => (
  <Stack sx={{ minWidth: 600 }}>
    {todos.map((todo, index) => (
      <TodoListItem key={index} todo={todo} />
    ))}
  </Stack>
);
