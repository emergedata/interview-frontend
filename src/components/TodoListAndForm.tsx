import { Box } from "@mui/material";
import { Todo } from "../models/todo";
import { NewTodoForm } from "./NewTodoForm";
import { TodoStack } from "./TodoStack";

export const TodoListAndFrom = ({ todos }: { todos: Todo[]; }) => (
  <Box
    sx={{
      display: "flex",
      flexDirection: "row",
      justifyContent: "center",
      alignItems: "start",
    }}
  >
    <TodoStack todos={todos} />
    <NewTodoForm />
  </Box>
);
