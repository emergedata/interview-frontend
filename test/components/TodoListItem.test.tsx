import { render } from "@testing-library/react";
import { TodoListItem } from "../../src/components/TodoListItem";
import { test, expect, describe } from "vitest";

const mockTodoData = {
  name: "Do something.",
  details: "Need to do something.",
};

describe("TodoListItem component", () => {
  test("renders component", () => {
    const result = render(<TodoListItem todo={mockTodoData} />);
    expect(result).toMatchSnapshot();
  });

  test("renders todo list item with name and details", () => {
    // Render the component with dummy mock todo data
    const screen = render(<TodoListItem todo={mockTodoData} />);

    expect(screen.findByText(mockTodoData.name)).toBeTruthy();
    expect(screen.findByText(mockTodoData.details)).toBeTruthy();
  });
});
