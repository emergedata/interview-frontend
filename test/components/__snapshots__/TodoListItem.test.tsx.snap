// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TodoListItem component > renders component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Do something.
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            Need to do something.
          </p>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          Do something.
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
        >
          Need to do something.
        </p>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
